hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/FieldOperations.java

```diff
@@ -594,6 +812,16 @@ public GenericRecord readGenericRecordOrPrimitive(GenericRecord genericRecord, S
         public void writeFieldFromRecordToWriter(DefaultCompactWriter writer, GenericRecord genericRecord, String fieldName) {
             writer.writeGenericRecord(fieldName, genericRecord.getGenericRecord(fieldName));
         }
+
+        @Override
+        public void writeJsonFormattedField(StringBuilder stringBuilder, AbstractGenericRecord record, String fieldName) {
+            GenericRecord genericRecord = record.getGenericRecord(fieldName);
+            if (genericRecord == null) {
+                stringBuilder.append("null");
+                return;
+            }
+            stringBuilder.append(genericRecord);
```

Comment:
```suggestion
            stringBuilder.append(record.getGenericRecord(fieldName));
```
Minor: This has the same effect